### ISSUE TITLE
[Remote-SSH Bug]: "Attach Visual Studio Code" fails for MacOS docker containers

### ISSUE THREAD

**Issue by junwha0511** [microsoft/vscode-remote-release#10037]
### Is there an existing issue for this bug?

- [X] I have searched the existing issues

### Required Troubleshooting Steps

- [X] I have followed these troubleshooting steps
- [X] I have tried both values of the `remote.SSH.useLocalServer` setting

### Connect Locally

It connects successfully

### ->

Hi, I'm trying to attach my VSCode to a docker container, and experiencing an issue.
It shows an error "Command failed". I guess it's from the incorrect replacement of username "qbit -n" since it should be "qbit".

![image](https://github.com/microsoft/vscode-remote-release/assets/17183234/b5458db4-5993-4df2-9f5a-888e8a2850a0)

`Error running command remote-containers.attachToRunningContainerFromViewlet: Command failed: mkdir -p /Users/qbit -n/.vscode-remote-containers/bin/8b3775030ed1a69b13e4f4c628c612102e30a681_1720348909009. This is likely caused by the extension that contributes remote-containers.attachToRunningContainerFromViewlet.`

### Expected Behavior

Opening a window on a docker container

### Steps To Reproduce

**Host:**: Windows 10 pro
**Server:**: macOS 14.5
**Container Image:**: arm64v8/ubuntu

### Remote-SSH Log

<details>
<summary>Remote-SSH Log</summary>
<p>

```
[19:50:26.355] Opening exec server for ssh-remote+7b22686f73744e616d65223a2253324c41422d4d6163227d
[19:50:26.428] Verified and reusing cached exec server for ssh-remote+7b22686f73744e616d65223a2253324c41422d4d6163227d
```

</p>
</details>

**Comment by bamurtaugh** [comment]
cc @chrmarti as well

**Comment by junwha0511** [comment]
@bamurtaugh FYI: An alternative way is to forward port 22 to host port xxxx and activate the SSH service inside a docker, and attach the VSCode to that port.

**Comment by junwha0511** [comment]
@chrmarti For which version was it resolved? I still have a same problem.

**Comment by chrmarti** [comment]
To verify on macOS:
- Enable `Remote Login` in the system settings.
- Connect to the local SSH server with `Remote-SSH: Connect to Host...` using `localhost`.
- Start a container to attach to: `docker run -d alpine sleep inf`
- Use `Dev Containers: Attach to Running Container...` to attach to the container.
- Verify this works.

**Comment by chrmarti** [comment]
@junwha0511 This will be in the next pre-release version. I will do that later today.

**Comment by chrmarti** [comment]
This is now available in Dev Containers 0.379.0-pre-release.
